documentation/bolt_examples.md

```diff
@@ -123,3 +123,226 @@ To convert the `restart_service.ps1` script to a task, giving you the ability to
+```
+
+4. From inside the project directory, install the Forge content by running: 
+
+```
+bolt puppetfile install
+```
+
+After it runs, you can see a `modules` directory inside the project directory, containing the modules you specified in the `Puppetfile`.
+
+### 3. Write a Bolt plan to apply Puppet code and orchestrate the deployment of a package resource using the Chocolatey provider. Plans allow you to run more than one task with a single command.
+
+1. Inside your project directory, in a file called `/plan/installer.pp`, create a plan called `puppet_chocolatey_tap::installer`. 
+
+The folder tree should look like this:
+
+```
+├──plan
+   └── installer.pp 
+```
+
+2. Copy the following code into the `installer.pp` file:
+
+```
+plan puppet_choco_tap::installer(
+   TargetSpec                                   $nodes,
+   String                                       $package,
+   Variant[Enum['absent', 'present'], String ]  $ensure = 'present',
+ ){
+
+ apply_prep($nodes)
+
+apply($nodes){
+  include chocolatey
+ }
+apply($nodes){
+  package { $package :
+    ensure    => $ensure,
+    provider  => 'chocolatey',
+    }
+  }
+}
+```
+
+Take note of the following features of the plan:
+
+- It has three parameters: the `TargetSpec`, a `package` string for the package name, and the `ensure` state of the package which allows for version, absent or present.
+- It has the `apply_prep` function call, which is used to install modules needed by `apply` on remote nodes as well as to gather facts about the nodes.
+- The first `apply` block installs the Chocolatey package manager. The Chocolatey provider is also deployed as a library with the Puppet agent in `apply_prep`.
+- The second `apply` block installs a specified package using the Chocolatey provider.
+
+2. To verify that the `puppet_chocolatey_tap` plan is available, run the following command: 
+
+```
+bolt plan show
+```
+
+The output looks like this:
+
+```
+facts
+facts::info
+puppet_choco_tap::installer
+puppetdb_fact
+```
+
+3. Run the plan with the `bolt plan run` command: 
+
+```
+bolt plan run puppet_choco_tap::installer package=frogsay --nodes=win --password
+```
+
+The output looks like this:
+
+```
+Starting: plan puppet_choco_tap::installer
+Starting: install puppet and gather facts on chocowin0.classroom.puppet.com, chocowin1.classroom.puppet.com
+Finished: install puppet and gather facts with 0 failures in 22.11 sec
+Starting: apply catalog on chocowin0.classroom.puppet.com, chocowin1.classroom.puppet.com
+Finished: apply catalog with 0 failures in 18.77 sec
+Starting: apply catalog on chocowin0.classroom.puppet.com, chocowin1.classroom.puppet.com
+Finished: apply catalog with 0 failures in 33.74 sec
+Finished: plan puppet_choco_tap::installer in 74.63 sec
+```
+
+4. To check that the installation worked, run the following `frogsay` command: 
+
+```        
+bolt command run 'frogsay ribbit' --targets windows --password
```

Comment:
Do not prompt for password. Given it is in the inventoryfile it would be ignored anyway.